controllers/cruisecontroltask_controller.go

```diff
@@ -292,28 +416,21 @@ func getActiveTasksFromCluster(instance *kafkav1beta1.KafkaCluster) *CruiseContr
 }
 
 // updateActiveTasks updates the state of the tasks from the CruiseControlTasksAndStates instance by getting their
-// status from CruiseControl using the provided scale.CruiseControlScaler.
-func updateActiveTasks(scaler scale.CruiseControlScaler, tasksAndStates *CruiseControlTasksAndStates) error {
-	tasks, err := scaler.GetUserTasks()
-	if err != nil {
-		return err
-	}
-
-	taskResultsByID := make(map[string]*scale.Result, len(tasks))
-	for _, task := range tasks {
-		taskResultsByID[task.TaskID] = task
+// status from CruiseControlOperation
+func updateActiveTasks(tasksAndStates *CruiseControlTasksAndStates, ccOperations []*banzaiv1alpha1.CruiseControlOperation) error {
+	ccOperationMap := make(map[string]*banzaiv1alpha1.CruiseControlOperation)
+	for i := range ccOperations {
+		operation := ccOperations[i]
+		ccOperationMap[operation.Name] = operation
```

Comment:
**Recommended**: I think we **SHOULD** do this in one step, and comment here a note to avoid accidental regression by someone hastily changing this into a direct assignment of an implicit iteration pointer instead of doing it this way.

```suggestion
		ccOperationMap[operation.Name] = ccOperations[i] // Note: deep copying pointer instead of iteration pointer intentionally because the implicit iteration pointer would point to a changing value.
```

Here the regular pointer assignment rules apply copying the pointer, the main thing is to not use the implicit iteration pointer because that has a changing underlying value.

Reply:
This is the last change request I posted, it is a minor thing.